v03_pipeline/lib/reference_data/compare_globals.py

```diff
@@ -0,0 +1,156 @@
+    get_ht_path,
+    import_ht_from_config_path,
+    parse_dataset_version,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class ReferenceDataGlobals:
+    paths: dict[str]
+    versions: dict[str]
+    enums: dict[str, dict[str, list[str]]]
+
+    def __init__(self, globals_struct: hl.Struct):
+        self.paths = self._struct_to_dict(globals_struct.paths)
+        self.versions = self._struct_to_dict(globals_struct.versions)
+        self.enums = self._struct_to_dict(globals_struct.enums)
+
+    def _struct_to_dict(self, struct: hl.Struct) -> dict:
+        result_dict = {}
+        for field in struct:
+            if isinstance(struct[field], hl.Struct):
+                result_dict[field] = self._struct_to_dict(struct[field])
+            else:
+                result_dict[field] = struct[field]
+        return result_dict
+
+
+def get_datasets_to_update(
+    joined_ht: hl.Table,
+    datasets: list[str],
+    reference_genome: ReferenceGenome,
+) -> list[str]:
+    datasets_to_update = []
+    for dataset in datasets:
+        if dataset not in joined_ht.row:
+            datasets_to_update.append(dataset)
+            continue
+
+        if not validate_joined_ht_globals_match_config(
+            joined_ht,
+            dataset,
+            reference_genome,
+        ):
+            datasets_to_update.append(dataset)
+    return datasets_to_update
+
+
+def validate_joined_ht_globals_match_config(
+    joined_ht: hl.Table,
+    dataset: str,
+    reference_genome: ReferenceGenome,
+) -> bool:
+    joined_ht_globals = ReferenceDataGlobals(hl.eval(joined_ht.index_globals()))
+    dataset_config = CONFIG[dataset][reference_genome.v02_value]
+    checks = {
+        'version': ht_version_matches_config(
+            joined_ht_globals,
+            dataset,
+            dataset_config,
+            reference_genome,
+        ),
+        'path': ht_path_matches_config(joined_ht_globals, dataset, dataset_config),
+        'enum': ht_enums_match_config(joined_ht_globals, dataset, dataset_config),
+        'select': ht_selects_match_config(joined_ht, dataset, dataset_config),
+    }
+
+    results = []
+    for check, result in checks.items():
+        if result is False:
+            logger.info(f'{check} mismatch for {dataset}')
+        results.append(result)
+    return all(results)
+
+
+def ht_version_matches_config(
+    joined_ht_globals: ReferenceDataGlobals,
+    dataset: str,
+    dataset_config: dict,
+    reference_genome: ReferenceGenome,
+) -> bool:
+    joined_ht_version = joined_ht_globals.versions.get(dataset)
+    if joined_ht_version is None:
+        return False
+
+    dataset_ht = import_ht_from_config_path(dataset_config, reference_genome)
+    try:
+        config_or_dataset_version = hl.eval(
+            parse_dataset_version(
+                dataset_ht,
+                dataset,
+                dataset_config,
+            ),
+        )
+    except HailUserError:
```

Comment:
hrm... I'm 🤔 hard on this one.... but I think we might actually want to just let the error ring out loudly here.  This basically means we've misconfigured one of the datasets (the path is pointing to a file with a mismatching version from what's in the config).